### DISCUSSION TITLE
Welcome to diagnostics Discussions!

### DISCUSSION THREAD

**Discussion by AlexeyAkhunov** [ledgerwatch/diagnostics#1] [Announcements]
## 👋 Welcome!

This discussion is a place to discuss the improvements to the diagnostics system project, whether you are doing it:
  * To participate in our search for potential Erigon developers - we ask you to propose an improvement to this repository as a Pull Request.
  * To perform part of our on-boarding program
  * To debug your node yourself, before you involve Erigon developers, and would like to add a missing useful feature.
  * For other reasons, for example, if you are working for the company who servers users with Erigon nodes and would like to utilise the diagnostics system to help their customers.

**Comment by shyam-patel-kira** [comment]
Hi Alexey, Is there any existing Erigon node live or a lighter version of the node that I can run locally to test the diagnostics system web app ?

**Reply by rashansmith** [reply]
Hey @shyam-patel-kira. My approach was to download the Erigon node code, and run the command: ```./build/bin/erigon --datadir <data_directory> --chain sepolia --internalcl --metrics```.  I believe running it on the Sepolia testnet and with the --internalcl flag makes it more lightweight. 

After running it once, it has been easy to spin up. I'm working on a PR to improve the documentation: https://github.com/ledgerwatch/diagnostics/pull/2 . If there is a faster way I'd love to add it!

**Reply by AlexeyAkhunov** [reply]
At the moment, I can only think of Sepolia, although my hope/dream is to use devnet tool eventually to simulate behaviour and also to use it as a demo for the diagnostics system

**Reply by shyam-patel-kira** [reply]
Thanks @rashansmith and @AlexeyAkhunov, I'll try spinning up sepolia testnet node for now

**Comment by rashansmith** [comment]
Hey Alexey, how can I execute transactions on the local Erigon node to spin up data in the diagnostics system as shown in the documentation for the [Reorg scanner](https://github.com/ledgerwatch/diagnostics#reorg-scanner) and the [block body](https://github.com/ledgerwatch/diagnostics#block-body-download). Is that necessary? Or is there a way to load sample data?

Here's what my views currently look like:
<img width="1286" alt="Screenshot 2023-04-27 at 12 24 05 AM" src="https://user-images.githubusercontent.com/6632748/234759520-d4882bae-7487-4b6f-a754-d9c789dc6a16.png">

<img width="900" alt="Screenshot 2023-04-27 at 12 17 13 AM" src="https://user-images.githubusercontent.com/6632748/234759618-29f65eb7-028c-4f61-9731-84be0f8d5ed1.png">

**Reply by AlexeyAkhunov** [reply]
In order to execute transactions you would need to have some SepoliaETH (I have some, but have no good way of distributing). It is not necessary to run reorg scanner, in your picture reorg scanner has completed without finding any reorgs. Block downloader monitoring is very raw, I used it on a particular stage of Sepolia sync, when it downloaded block bodies, after that is done, there is no much to see there. Again, I was hoping to use devnet tool to simulate both reorgs and lots of block bodies to download, but this will surely take time, and I thought there are lots of other improvements we can work on before that.

**Reply by rashansmith** [reply]
Thank you for explaining. For the reorg scanner, one of the idea tasks involves adding the ability to click on block numbers to get more info on that reorg. It would be great to test it visually, unless it's not needed or can be tested by someone else with an active node. I also have some sepoliaETH but wasn't sure how to connect it to the node.

**Reply by AlexeyAkhunov** [reply]
I previously used Mainnet node or Gnosis chain node that I have been running for a while (which does produce reorgs) to test the reorg scanner. Eventually, I would like to use it to confirm some theories about the reorgs on Polygon.
I do agree that it is hard to visually test it at the moment, and that tells me that my next effort should be spent on bringing the devnet tool up to speed to enable that.

**Comment by kokonutwh1skey** [comment]
Hi Alexey, I just can't come up with a ideal way to integrate `lumberjack` into the log. 
I'm thinking about fetching logs every X seconds and writing them to a local dir or a memory cache and having web page read from it. Any suggestion?

**Reply by AlexeyAkhunov** [reply]
I have only integrated lumberjack once, recently for Erigon: https://github.com/ledgerwatch/erigon/blob/8a3b9e6d17c0f9c7ef9bf58d0d10c1246303e548/turbo/logging/logging.go#L148

It does imply using the same logging library as Erigon, but it should be ok

**Reply by AlexeyAkhunov** [reply]
To clarify, the integration of lumberjack was not suggested for the functionality of remotely viewing logs from Erigon nodes, but simply for the diagnostics system's own logs :)

**Reply by kokonutwh1skey** [reply]
I see. I thought it is for remotely viewing logs from Erigon nodes :)

**Comment by anshulforyou** [comment]
Hey Alexey, I am trying to connect my local erigon node to the diagnostics system but every time I run the command mentioned in the readme `./build/bin/erigon support --metrics.urls http://metrics.addr:metrics.port/debug/metrics --diagnostics.url https://localhost:8080/support/47779410 --insecure` and use the diagnostic's ui to access the command line, I get this error `ERROR: Requesting metrics url [http://localhost:6060/debug/metrics], query [/cmdline], err: Get "http://localhost:6060/debug/metrics/cmdl
ine": dial tcp [::1]:6060: connect: connection refused`

Is there something I am missing?

<img width="1329" alt="Screenshot 2023-05-20 at 6 45 49 PM" src="https://github.com/ledgerwatch/diagnostics/assets/55681256/9cfc1bed-13e0-4165-86b9-c455a384b7ba">

**Reply by acuderman** [reply]
Hi, I don't know if this is the issue, but I can reproduce the same error if I start the erigon client without "--metrics" argument. Using the following command works for me: 
`./build/bin/erigon --datadir <data_directory> --chain sepolia --internalcl --metrics`

**Reply by anshulforyou** [reply]
Yes, it helped. I think there are some minor errors in the readme. Made a [PR](https://github.com/ledgerwatch/diagnostics/pull/14) for it!

**Comment by anshulforyou** [comment]
Hey Alexey, for "Adding a monitoring for header downloader" point in the "Ideas for possible improvements" section. What exact parameters of the HeaderDownload struct should be monitored by the diagnostics system?
